python/kserve/kserve/protocol/rest/v2_endpoints.py

```diff
@@ -128,7 +128,7 @@ async def infer(
                                    data=input.data,
                                    parameters={} if input.parameters is None else input.parameters
                                    ) for input in request_body.inputs]
-        infer_request = InferRequest(model_name=model_name, infer_inputs=infer_inputs,
+        infer_request = InferRequest(id=request_body.id, model_name=model_name, infer_inputs=infer_inputs,
```

Comment:
seems like it is not part of the constructor :(, @markgeejw can you help add it?
```
FAILED kserve/test/test_server.py::TestTFHttpServer::test_infer - TypeError: InferRequest.__init__() got an unexpected keyword argument 'id'
FAILED kserve/test/test_server.py::TestRayServer::test_infer - TypeError: InferRequest.__init__() got an unexpected keyword argument 'id'
```

Reply:
Alright I just used the request_id in the constructor instead